src/web/src/app/space/list/component.html

```diff
@@ -0,0 +1,122 @@
+						<mat-hint i18n="">Пространство</mat-hint>
+					</mat-form-field>
+				</div> 
+				<div class="left-float left-padding-10">
+					<mat-form-field appearance="fill">
+						<input matInput type="text" formControlName="name" required />
+						<mat-hint i18n="">Имя ссылки</mat-hint>
+					</mat-form-field>
+				</div>
+				<div class="left-float left-padding-10">
+					<mat-form-field appearance="fill">
+						<input matInput type="date" formControlName="expiredAt" required />
+						<mat-hint i18n="">Срок действия</mat-hint>
+					</mat-form-field>
+				</div>
+				<div class="left-float left-padding-10"><button mat-icon-button type="submit" i18n=""><mat-icon>done</mat-icon></button></div>
+				<div class="left-float left-padding-10"><button mat-icon-button (click)="onLinkFormCloseBtn()"><mat-icon>close</mat-icon></button></div>
+			</form>
+		</div> <!-- end form for creating link -->
+		
+		<!-- listing links -->
+		<div *ngFor="let link of receivedLinks" class="float-row-with-padding">
+			<div class="left-float">{{getSpaceNameById(link.spaceId)}}</div>
+			<div class="left-float left-padding-10">{{link.name}}</div>
+			<div class="left-float left-padding-10">{{link.expiredAt | date}}</div>
+			<div class="left-float left-padding-10"><button mat-raised-button i18n="@@SPACES.LIST.REFS.COPYBTN" (click)="onLinkCopyBtn(link)">Копировать</button></div>
+			<div class="left-float left-padding-10" (click)="onLinkDelBtn(link)"><button mat-icon-button><mat-icon>close</mat-icon></button></div>
+		</div>
+	</div>
+
+	<mat-paginator #linksPaginator *ngIf="links.length > linksPageSize" class="width-and-clear" [length]="links.length" [pageSize]="linksPageSize" (page)="pageEvent = getPaginatorLinks($event)"></mat-paginator> 
+	<!-- end of listing links -->
+</ng-container>
+
+<!-- listing spaces -->
+<hr class="width-and-clear">
+<div class="spaces-heading">
+	<h2 i18n="@@SPACES.LIST.SPACES.HEADING">Пространства</h2>
+	<a routerLink="/space/add"><button class="mat-icon-button"><mat-icon>add</mat-icon></button></a>
+</div>
+<div class="width-and-clear">
+	<div *ngFor="let space of receivedSpaces" class="float-row-with-padding">
+		<div class="left-float"><a routerLink="/space/1">{{space.name}}</a></div>
+		<div class="left-float left-padding-10">{{space.key}}</div>
+		<div class="left-float left-padding-10"><button mat-icon-button (click)="onSpaceInviteAddUser(space)"><mat-icon>person_add</mat-icon></button></div>
+		<div class="left-float left-padding-10"><button mat-icon-button (click)="onSpaceAddLink(space)"><mat-icon>link</mat-icon></button></div>
+		<div class="left-float left-padding-10"><button mat-icon-button (click)="onSpaceDelBtn(space)"><mat-icon>close</mat-icon></button></div>
+	</div>
+	<mat-paginator #spacesPaginator *ngIf="spaces.length > spacesPageSize" class="width-and-clear" [length]="spaces.length" [pageSize]="spacesPageSize" (page)="pageEvent = getPaginatorSpaces($event)"></mat-paginator> 
```

Comment:
length - общее кол-во придёт с бэка. И поскольку ты используешь ViewChild, его наверное сразу туда можно будет установить.

Ниже вариант getSpaces

```suggestion
	<mat-paginator #spacesPaginator class="width-and-clear"
	*ngIf="spaces.length > spacesPageSize"
	[pageSize]="spacesPageSize"
	(page)="getSpaces($event.pageIndex, $event.pageSize)"></mat-paginator> 
```

Reply:
Сходу length в пагинотор на стороне компонента поместить не получилось. Вернусь к этому завтра по плану.

Reply:
Пробовал ещё раз обращаться к пагинатору из компонента в ngOnInit и в ngAfterViewInit - в обеих случаях пагинатор ещё не существует. 

Что касается общего количества, то ставлю пока через переменную, заполняемую в момент получения данных из ангуларовского сервиса.

Reply:
Видимо тут лучше через переменную как ты сделал. Проверил почему не сработало без неё, это потому что у нас весь список вместе с пагинатором скрытый когда список пуст. А отображается только после того как список запрошен уже в первый раз. Если бы мы его не скрывали, то тогда бы всё получилось, по тому что мы обращаемся к пагинатору не в ngOnInit, а после события получения списка с сервера, т.е. пока идёт запрос к серверу, пагинатор уже инициализируется.